File: app/src/main/java/com/androidessence/pinchzoomtextview/sample/MainActivity.java
package com.androidessence.pinchzoomtextview.sample;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.androidessence.pinchzoomtextview.PinchZoomTextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final PinchZoomTextView textView = (PinchZoomTextView) findViewById(R.id.text_view);

        final Button enableToggle = (Button) findViewById(R.id.zoom_toggle);
        enableToggle.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                boolean newEnabledState = !textView.isZoomEnabled();
                textView.setZoomEnabled(newEnabledState);
                enableToggle.setText(
                        newEnabledState ? getString(R.string.zoom_enabled) : getString(R.string.zoom_disabled)
                );
            }
        });
    }
}

File: app/src/main/java/com/androidessence/pinchzoomtextview/sample/MainActivity.kt
package com.androidessence.pinchzoomtextview.sample

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.androidessence.pinchzoomtextview.PinchZoomTextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val textView = findViewById<PinchZoomTextView>(R.id.text_view)
        val zoomToggle = findViewById<TextView>(R.id.zoom_toggle)

        zoomToggle.setOnClickListener {
            val newEnabledState = !textView.zoomEnabled
            textView.zoomEnabled = newEnabledState
            zoomToggle.text = if (newEnabledState) getString(R.string.zoom_enabled) else getString(R.string.zoom_disabled)
        }
    }
}